chains/solana/contracts/programs/example-burnmint-token-pool/src/common.rs

```diff
@@ -0,0 +1,521 @@
+use anchor_lang::prelude::*;
```

Comment:
nit:
What about the common files to be in a separate module? `common.rs` & `rate_limiter.rs`?

Reply:
i can give it a try - but my main concern is that it potentially breaks some of the anchor idl + go binding generation.

let me give it a shot at reorganization and I can let you know :)

Reply:
i split this into a separate "null program" (no functionality) so it exports types cleanly since crates don't work with IDL/gobinding generation

now the pools all import `base-token-pool` rather than the `burn/mint` pool